src/topology/constructions.lean

```diff
@@ -284,6 +284,25 @@ is_open.inter (hs.preimage continuous_fst) (ht.preimage continuous_snd)
 lemma nhds_prod_eq {a : α} {b : β} : 𝓝 (a, b) = 𝓝 a ×ᶠ 𝓝 b :=
 by rw [filter.prod, prod.topological_space, nhds_inf, nhds_induced, nhds_induced]
 
+/-- If a function `f x y` is such that `y ↦ f x y` is continuous for all `x`, and `x` lives in a
+discrete space, then `f` is continuous. -/
+lemma continuous_uncurry_of_discrete_topology [discrete_topology α]
+  {f : α → β → γ} (hf : ∀ a, continuous (f a)) : continuous (function.uncurry f) :=
```

Comment:
This proof felt needlessly complicated, but to be honest my trick to simplify it is to add a filter lemma that is rather ad hoc. Feel free to take it or not.
```lean
lemma filter.map_pure_prod {α β γ : Type*} (f : α → β → γ) (a : α) (B : filter β) :
  filter.map (function.uncurry f) (pure a ×ᶠ B) = filter.map (f a) B :=
begin
  ext u,
  simp only [mem_map],
  rw ((has_basis_pure a).prod (basis_sets B)).mem_iff,
  simp only [←preimage_comp, true_and, singleton_prod, image_subset_iff, exists_prop, 
             prod.exists, exists_const],
  simp_rw [← exists_prop], 
  exact exists_mem_subset_iff
end

/-- If a function `f x y` is such that `y ↦ f x y` is continuous for all `x`, and `x` lives in a
discrete space, then `f` is continuous. -/
lemma continuous_uncurry_of_discrete_topology [discrete_topology α]
  {f : α → β → γ} (hf : ∀ a, continuous (f a)) : continuous (function.uncurry f) :=
begin
  apply continuous_iff_continuous_at.2,
  rintros ⟨a, x⟩,
  change map _ _ ≤ _,
  rw [nhds_prod_eq, nhds_discrete, filter.map_pure_prod],
  exact (hf a).continuous_at
end
```

Reply:
To golf the proof your new lemma, you can use `filter.pure_prod`.